### ISSUE TITLE
osc ls -b can't cope with Unicode

### ISSUE THREAD

**Issue by jengelh** [openSUSE/osc#1018]
```
(api.opensuse.org)

$ osc ls -b home:jengelh/test
Traceback (most recent call last):
  File "/usr/bin/osc", line 45, in <module>
    r = babysitter.run(osccli)
  File "/usr/lib/python3.8/site-packages/osc/babysitter.py", line 67, in run
    return prg.main(argv)
  File "/usr/lib/python3.8/site-packages/osc/cmdln.py", line 347, in main
    return self.cmd(args)
  File "/usr/lib/python3.8/site-packages/osc/cmdln.py", line 370, in cmd
    retval = self.onecmd(argv)
  File "/usr/lib/python3.8/site-packages/osc/cmdln.py", line 504, in onecmd
    return self._dispatch_cmd(handler, argv)
  File "/usr/lib/python3.8/site-packages/osc/cmdln.py", line 1235, in _dispatch_cmd
    return handler(argv[0], opts, *args)
  File "/usr/lib/python3.8/site-packages/osc/commandline.py", line 392, in do_list
    results.append((repo, get_binarylist(apiurl, project, repo.name, repo.arch, package=package, verbose=opts.verbose)))
  File "/usr/lib/python3.8/site-packages/osc/core.py", line 5719, in get_binarylist
    f = http_GET(u)
  File "/usr/lib/python3.8/site-packages/osc/core.py", line 3413, in http_GET
    def http_GET(*args, **kwargs):    return http_request('GET', *args, **kwargs)
  File "/usr/lib/python3.8/site-packages/osc/core.py", line 3404, in http_request
    return urlopen(req)
  File "/usr/lib64/python3.8/urllib/request.py", line 222, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/lib64/python3.8/urllib/request.py", line 525, in open
    response = self._open(req, data)
  File "/usr/lib64/python3.8/urllib/request.py", line 542, in _open
    result = self._call_chain(self.handle_open, protocol, protocol +
  File "/usr/lib64/python3.8/urllib/request.py", line 502, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.8/site-packages/osc/oscssl.py", line 222, in https_open
    h.request(req.get_method(), request_uri, req.data, headers)
  File "/usr/lib64/python3.8/http/client.py", line 1256, in request
    self._send_request(method, url, body, headers, encode_chunked)
  File "/usr/lib64/python3.8/http/client.py", line 1267, in _send_request
    self.putrequest(method, url, **skips)
  File "/usr/lib64/python3.8/http/client.py", line 1105, in putrequest
    self._output(self._encode_request(request))
  File "/usr/lib64/python3.8/http/client.py", line 1185, in _encode_request
    return request.encode('ascii')
UnicodeEncodeError: 'ascii' codec can't encode character '\xa0' in position 26: ordinal not in range(128)
```

**Comment by dmach** [comment]
What is the expected result?
I have an open pull request (https://github.com/openSUSE/osc/pull/1000) that uses urllib3 to make http requests and it ends like this:
```
DEBUG: GET https://api.opensuse.org/source/home:jengelh/_meta
DEBUG: GET https://api.opensuse.org/build/home:jengelh/st andard/x86_64/test
DEBUG: GET https://api.opensuse.org/build/home:jengelh/sta$n/x86_64/test
DEBUG: GET https://api.opensuse.org/build/home:jengelh/sta ndard/x86_64/test
DEBUG: GET https://api.opensuse.org/build/home:jengelh/st?oh=no&anyway/x86_64/test

Server returned an error: HTTP Error 400: unknown parameter 'anyway/x86_64/test', 'oh'
unknown parameter 'anyway/x86_64/test', 'oh'
```

**Comment by jengelh** [comment]
Expected result: Don't throw a Python exception, show something else (e.g. HTTP 200/4xx/5xx result).

**Comment by dmach** [comment]
The OBS part was closed as a duplicate of https://github.com/openSUSE/open-build-service/issues/7021
It seems to be a known issue for some time.
My pull request fixes only the client part.
We may add more illegal characters to the list in the future, but besides that, that's probably all we can do in osc now.

**Comment by dmach** [comment]
The problem on osc side was fixed in #1000 already.
